azurerm/internal/services/datafactory/resource_arm_data_factory_linked_service_blob_storage.go

```diff
@@ -0,0 +1,255 @@
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMDataFactoryLinkedServiceDatasetName,
+			},
+
+			"data_factory_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.DataFactoryName(),
+			},
+
+			// There's a bug in the Azure API where this is returned in lower-case
+			// BUG: https://github.com/Azure/azure-rest-api-specs/issues/5788
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
+
+			"connection_string": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"integration_runtime_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"parameters": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"annotations": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"additional_properties": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceArmDataFactoryLinkedServiceBlobStorageCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataFactory.LinkedServiceClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	dataFactoryName := d.Get("data_factory_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, dataFactoryName, name, "")
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Data Factory Linked Service BlobStorage Anonymous %q (Data Factory %q / Resource Group %q): %+v", name, dataFactoryName, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_data_factory_linked_service_blob_storage", *existing.ID)
+		}
+	}
+
+	connectionString := d.Get("connection_string").(string)
+	secureString := datafactory.SecureString{
+		Value: &connectionString,
+		Type:  datafactory.TypeSecureString,
+	}
+
+	blobProperties := &datafactory.AzureBlobStorageLinkedServiceTypeProperties{
+		ConnectionString: &secureString,
+	}
+
+	description := d.Get("description").(string)
+
+	blobStorageLinkedService := &datafactory.AzureBlobStorageLinkedService{
+		Description: &description,
+		AzureBlobStorageLinkedServiceTypeProperties: blobProperties,
+		Type: datafactory.TypeAzureBlobStorage,
+	}
```

Comment:
Could we combine this into a single initialization?
```suggestion
	blobStorageLinkedService := &datafactory.AzureBlobStorageLinkedService{
		Description: utils.String(d.Get("description").(string)),
		AzureBlobStorageLinkedServiceTypeProperties: datafactory.AzureBlobStorageLinkedServiceTypeProperties{
			ConnectionString: &datafactory.SecureString{
				Value: utils.String(d.Get("connection_string").(string)),
				Type:  datafactory.TypeSecureString,
			},
		},
		Type: datafactory.TypeAzureBlobStorage,
	}
```